### PR TITLE
udev-builtin-input_id: any i2c mouse is a pointing stick

### PR DESCRIPTION
**Pull request by whot**
Where we have a device that looks like a mouse and is connected over i2c, tag
it as pointing stick. There is no such thing as a i2c mouse.

Even touchpads that aren't recognized by the kernel will not show up as i2c
mouse - either the touchpad follows the Win8.1 specs in which case the kernel
switches it to multitouch mode and it shows up like a touchpad. The built-in
trackpoint, if any, is then the i2c mouse device.

Where the touchpad doesn't follow the spec, the kernel will not handle it and
the touchpad remains on the PS/2 legacy bus - not i2c. Hence we can assume
that any i2c mouse device is really a pointing stick.

cc @bentiss

edit: originally reported against libinput here: https://gitlab.freedesktop.org/libinput/libinput/issues/423

### CODE REVIEW

**Comment by whot** [comment]
@bentiss - reading through this again I'm wondering why we wouldn't implement exactly this in the hid-multitouch driver in the kernel and assign `INPUT_PROP_POINTINGSTICK` directly.

File: src/udev/udev-builtin-input_id.c [hunk 1]
```diff
@@ -123,8 +123,25 @@ static void get_cap_mask(sd_device *pdev, const char* attr,
         }
 }
 
+static struct input_id get_input_id(sd_device *dev) {
+    const char *v;
```
Comment (keszybz) [inline src/udev/udev-builtin-input_id.c]: Please make 8 space indentation.

File: src/udev/udev-builtin-input_id.c [hunk 2]
```diff
@@ -123,8 +123,25 @@ static void get_cap_mask(sd_device *pdev, const char* attr,
...
+    struct input_id id = {0};
```
Comment (keszybz) [inline src/udev/udev-builtin-input_id.c]: Just {}.

File: src/udev/udev-builtin-input_id.c [hunk 3]
```diff
@@ -123,8 +123,25 @@ static void get_cap_mask(sd_device *pdev, const char* attr,
...
+
+    if (sd_device_get_sysattr_value(dev, "id/bustype", &v) >= 0)
+        id.bustype = strtoul(v, NULL, 16);
```
Comment (keszybz) [inline src/udev/udev-builtin-input_id.c]: I'd prefer
```C
(void) safe_atoux16(v, &id.bustype);
```

**Comment by whot** [comment]
all fixed as requested

**Comment by keszybz** [comment]
semaphoreci fails with "ERROR: Invalid signature for /tmp/tmp.ToMGWiv0OL/index.asc", whatever that means.

**Pull request merged**